userspace/libsinsp/event.cpp

```diff
@@ -1197,33 +1197,37 @@ Json::Value sinsp_evt::get_param_as_json(uint32_t id, OUT const char** resolved_
 			const struct ppm_name_value *flags = (const struct ppm_name_value *)m_info->params[id].info;
 			uint32_t initial_val = val;
 
-			while(flags != NULL && flags->name != NULL && flags->value != initial_val)
+			while(flags != NULL && flags->name != NULL)
 			{
-				if ((exact_match && flags->value == initial_val) ||
+				bool match = false;
+				if (exact_match)
+				{
+					match = flags->value == initial_val;
+				}
+				else
+				{
 					// If flag is 0, then initial_val needs to be 0 for the flag to be resolved
-					((flags->value == 0 && initial_val == 0) ||
-				     (flags->value != 0 && (val & flags->value) == flags->value && val != 0)))
+					if ((flags->value == 0 && initial_val == 0) ||
+					   (flags->value != 0 && (val & flags->value) == flags->value && val != 0))
+					{
+						match = true;
+						// We remove current flags value to avoid duplicate flags e.g. PPM_O_RDWR, PPM_O_RDONLY, PPM_O_WRONLY
+						val &= ~flags->value;
+					}
+				}
```

Comment:
Maybe this would be clearer a bit still if we separated the two cases? i.e.
```
if (flags->value == 0 && initial_val == 0)
{
	// If flag is 0, then initial_val needs to be 0 for the flag to be resolved
	match = true;
}
else if (flags->value != 0 && (val & flags->value) == flags->value && val != 0)
{
	match = true;
	// We remove current flags value to avoid duplicate flags e.g. PPM_O_RDWR, PPM_O_RDONLY, PPM_O_WRONLY
	val &= ~flags->value;
}
```

Reply:
We can do that sure! I don't think this improves readability though :)